sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/channel/ManagedChannelFactory.java

```diff
@@ -98,7 +101,23 @@ public ManagedChannel forDescriptor(ApiServiceDescriptor apiServiceDescriptor) {
     if (directExecutor) {
       channelBuilder = channelBuilder.directExecutor();
     }
-    return channelBuilder.build();
+    ManagedChannel channel = channelBuilder.build();
+    toClose.add(channel);
+    return channel;
+  }
+
+  public void shutdown() {
+    for (ManagedChannel channel : toClose) {
+      channel.shutdownNow();
+    }
+    for (ManagedChannel channel : toClose) {
+      try {
+        channel.awaitTermination(5, TimeUnit.SECONDS);
+      } catch (InterruptedException exn) {
+        System.err.println("Timed out closing channel: " + exn);
```

Comment:
Should we use a `org.slf4j.Logger` and `LOG.warn("Timed out closing channel: {}", exn")`.  Feel free to resolve this comment if you disagree.